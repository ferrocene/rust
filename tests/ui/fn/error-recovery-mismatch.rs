//@ revisions: edition2015 edition2018
//@ [edition2015] edition: 2015
//@ [edition2018] edition: 2018..
// Used to ICE due to a size mismatch between the actual fake signature of `fold` and the
// generated signature used reporting the parameter mismatch at the call site.
// See issue #135124

trait A  {
    fn y(&self)
    {
        fn call() -> impl Sized {}
        self.fold(call(), call());
    }
    fn fold<T>(&self, _: T, &self._) {}
    //[edition2018]~^ ERROR expected one of `:` or `|`, found `)`
    //~^^ ERROR unexpected `self` parameter in function
    //~| ERROR expected one of `)` or `,`, found `.`
    //~| ERROR identifier `self` is bound more than once in this parameter list
    //[edition2015]~| WARNING anonymous parameters are deprecated
    //[edition2015]~| WARNING this is accepted in the current edition
    //[edition2015]~| ERROR the placeholder `_` is not allowed within types
}

fn main() {}
