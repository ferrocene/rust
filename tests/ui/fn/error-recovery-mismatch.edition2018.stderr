error: unexpected `self` parameter in function
  --> $DIR/error-recovery-mismatch.rs:14:29
   |
LL |     fn fold<T>(&self, _: T, &self._) {}
   |                             ^^^^^ must be the first parameter of an associated function

error: expected one of `:` or `|`, found `)`
  --> $DIR/error-recovery-mismatch.rs:14:36
   |
LL |     fn fold<T>(&self, _: T, &self._) {}
   |                                    ^ expected one of `:` or `|`
   |
   = note: anonymous parameters are removed in the 2018 edition (see RFC 1685)
help: explicitly ignore the parameter name
   |
LL |     fn fold<T>(&self, _: T, &self._: _) {}
   |                                   ++

error: expected one of `)` or `,`, found `.`
  --> $DIR/error-recovery-mismatch.rs:14:34
   |
LL |     fn fold<T>(&self, _: T, &self._) {}
   |                                  ^
   |                                  |
   |                                  expected one of `)` or `,`
   |                                  help: missing `,`

error[E0415]: identifier `self` is bound more than once in this parameter list
  --> $DIR/error-recovery-mismatch.rs:14:30
   |
LL |     fn fold<T>(&self, _: T, &self._) {}
   |                              ^^^^ used as parameter more than once

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0415`.
